### PR TITLE
Improved lexer splitting

### PR DESCRIPTION
**Pull request by szarnekow**

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java [hunk 1]
```diff
@@ -213,6 +222,41 @@ private static String getVarnameFromDecl(String varDecl) {
 			return null;
 		return m.group(1);
 	}
+	
+	void copyTailAndRefactorDfaTransitionS() {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java]: package private by intention. if so why not private as the caller?

Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java]: good catch

Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java]: I copied it from `boolean copyUntilMethod()` ... will make both private

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java [hunk 2]
```diff
@@ -34,6 +34,8 @@
 	public static final Pattern OUTER_SWITCH_PATTERN = Pattern.compile("^\\s{8}switch", 0);
 	public static final Pattern OUTER_IF_PATTERN = Pattern.compile("^\\s{8}if", 0);
 	public static final Pattern OUTER_BRACE_IN_IF_CASCADE_PATTERN = Pattern.compile("^(\\s{12}[^\\s]+.*)}\\s*$", 0);
+	public static final Pattern DFA_TRANSITIONS_PATTERN = Pattern.compile("(static final String\\[\\] (DFA\\d+_transitionS) = )\\{", 0);
+	public static final Pattern DFA_FIELD_END_PATTERN = Pattern.compile("^\\s{4}};", 0);
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java]: sounds very stable.

Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java]: Irony?

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java [hunk 3]
```diff
@@ -213,8 +222,43 @@ private static String getVarnameFromDecl(String varDecl) {
 			return null;
 		return m.group(1);
 	}
+	
+	private void copyTailAndRefactorDfaTransitionS() {
```
[the remark this answers is not in the document]
Comment (szarnekow) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/AntlrLexerSplitter.java]: The original field name is smth like DFA123_transitionS - that's why the name.

**Pull request merged**